schema.c

```diff
@@ -445,9 +479,60 @@ sch_load_namespace_mappings (sch_instance *instance, const char *filename)
     }
 }
 
+static void
+sch_load_model_list (sch_instance *instance, const char *path, const char *model_list_filename)
+{
+    FILE *fp = NULL;
+    char *name;
+    char buf[256];
+
+    if (!instance)
+        return;
+
+    name = g_strdup_printf ("%s/%s", path, model_list_filename);
+    fp = fopen (name, "r");
+    if (fp)
+    {
+        if (!instance->model_hash_table)
+            instance->model_hash_table = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
+
+        while (fgets (buf, sizeof (buf), fp) != NULL)
```

Comment:
while this is safe enough, it doesn't cater for lines longer than 256. will we have enough control of the list of model names to be sure that this will be the case?

Reply:
Fixed